vim-plug/plugins.vim

```diff
@@ -178,3 +178,26 @@ autocmd VimEnter *
   \  if len(filter(values(g:plugs), '!isdirectory(v:val.dir)'))
   \|   PlugInstall --sync | q
   \| endif
+
+" Run updates every week automatically on startup
+function! OnVimEnter() abort
+  if exists('g:plug_home')
+    let l:filename = printf('%s/.vim_plug_update', g:plug_home)
+    if filereadable(l:filename) == 0
+      call writefile([], l:filename)
+    endif
+
+    let l:this_week = strftime('%Y_%V')
+    let l:contents = readfile(l:filename)
+    if index(l:contents, l:this_week) < 0
+      call execute('PlugUpgrade')
+      call execute('PlugUpdate')
+      call execute('CocUpdate')
+      call execute('UpdateRemotePlugins')
+      call execute('PlugClean')
```

Comment:
I do not know why but PlugClean throws errors in this scenario, better to remove line 197.